internal/featureflags/featureflag.go

```diff
@@ -0,0 +1,55 @@
+package featureflags
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	featurePrefix    = "FEATURE_"
+	featureSeparator = "="
+	//nolint:stylecheck
+	FeatureOIDC = "FEATURE_PREVIEW_OIDC_DB_ACCESS"
```

Comment:
Is this the only known feature flag so far?
If so. How do we know we will need this feature flags going forward for more values?

Reply:
Actually this could be what we need if we go for a more controlled process for releasing features. Instead of a commit releasing a new feature or behaviour directly, a flag might control whether some of those are active and maybe even depend on what version value it is being built, the default feature flag value is different. One single codebase can release different operator versions without branching.

Reply:
Yes, this is the only feature flag so far. I implemented then as env variables as it is easier to configure for the k8s `Deployment` and/or `Pod` resource. We can also add flag processing if needed (e.g. `--features=featureA,featureB`.